src/components/BulkEditAccessPoints/components/BulkEditAPTable/index.js

```diff
@@ -6,64 +6,36 @@ import { EditableRow } from './components/EditableRow';
 import { EditableCell } from './components/EditableCell';
 import styles from './index.module.scss';
 
-const BulkEditAPTable = ({
-  tableColumns,
-  tableData,
-  onEditedRows,
-  onLoadMore,
-  isLastPage,
-  resetEditedRows,
-}) => {
+const BulkEditAPTable = ({ tableColumns, tableData, onEditedRows, onLoadMore, isLastPage }) => {
   const [bulkEditTableData, setTableData] = useState([]);
   const [editedRows, setEditedRows] = useState([]);
+
+  useEffect(() => {
+    onEditedRows(editedRows);
+  }, [editedRows]);
+
+  useEffect(() => {
+    setTableData(tableData);
+  }, [tableData]);
+
   const components = {
     body: {
       row: EditableRow,
       cell: EditableCell,
     },
   };
 
-  useEffect(() => {
-    setTableData(tableData);
-  }, [tableData]);
-
-  let tempEditedRows = [];
-
-  const handleSave = row => {
-    const newData = [...bulkEditTableData];
-    const channel = JSON.parse(`[${row.channel}]`);
-    const cellSize = JSON.parse(`[${row.cellSize}]`);
-    const probeResponseThreshold = JSON.parse(`[${row.probeResponseThreshold}]`);
-    const clientDisconnectThreshold = JSON.parse(`[${row.clientDisconnectThreshold}]`);
-    const snrDrop = JSON.parse(`[${row.snrDrop}]`);
-    const minLoad = JSON.parse(`[${row.minLoad}]`);
-    const updatedRow = {
-      ...row,
-      channel,
-      cellSize,
-      probeResponseThreshold,
-      clientDisconnectThreshold,
-      snrDrop,
-      minLoad,
-    };
-    const itemIndex = editedRows.findIndex(a => row.key === a.key);
-    const tempRows = [...editedRows];
+  const handleSave = (row, values) => {
+    const updatedRow = { ...row };
+    const key = Object.keys(values)[0];
+    updatedRow[key] = JSON.parse(`[${values[key]}]`);
 
-    if (itemIndex === -1) {
-      tempRows.push(updatedRow);
-      setEditedRows(tempRows);
+    if (!editedRows.some(a => row.key === a.key)) {
```

Comment:
just make editedRows a key, value object then you don't have to do all this array handling